src/main/java/org/team1540/robot2018/subsystems/DriveTrain.java

```diff
@@ -0,0 +1,105 @@
+package org.team1540.robot2018.subsystems;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
+import org.team1540.base.ChickenSubsystem;
+import org.team1540.base.wrappers.ChickenTalon;
+import org.team1540.robot2018.RobotMap;
+import org.team1540.robot2018.Tuning;
+import org.team1540.robot2018.commands.JoystickDrive;
+
+public class DriveTrain extends ChickenSubsystem {
+
+  private ChickenTalon left = new ChickenTalon(RobotMap.DRIVE_LEFT);
+  private ChickenTalon left2 = new ChickenTalon(RobotMap.DRIVE_LEFT_2);
+  private ChickenTalon left3 = new ChickenTalon(RobotMap.DRIVE_LEFT_3);
+  private ChickenTalon[] lefts = new ChickenTalon[]{left, left2, left3};
+  private ChickenTalon right = new ChickenTalon(RobotMap.DRIVE_RIGHT);
+  private ChickenTalon right2 = new ChickenTalon(RobotMap.DRIVE_RIGHT_2);
+  private ChickenTalon right3 = new ChickenTalon(RobotMap.DRIVE_RIGHT_3);
+  private ChickenTalon[] rights = new ChickenTalon[]{right, right2, right3};
+  private ChickenTalon[] talons = new ChickenTalon[]{left, left2, left3, right, right2, right3};
+  private ChickenTalon[] masters = new ChickenTalon[]{left, right};
+
+  public DriveTrain() {
+    this.add(talons);
+    this.setPriority(10);
+
+    left.configSelectedFeedbackSensor(FeedbackDevice.QuadEncoder);
+    right.configSelectedFeedbackSensor(FeedbackDevice.QuadEncoder);
+
+    left.setSensorPhase(true);
+
+    for (ChickenTalon talon : lefts) {
+      talon.setInverted(false);
+    }
+
+    right.setSensorPhase(true);
+
+    for (ChickenTalon talon : rights) {
+      talon.setInverted(true);
+    }
+
+    left2.set(ControlMode.Follower, left.getDeviceID());
+    left3.set(ControlMode.Follower, left.getDeviceID());
+
+    right2.set(ControlMode.Follower, right.getDeviceID());
+    right3.set(ControlMode.Follower, right.getDeviceID());
+
+    for (ChickenTalon talon : talons) {
+      talon.setBrake(true);
+    }
+  }
+
+  @Override
+  public void initDefaultCommand() {
+    setDefaultCommand(new JoystickDrive());
+  }
+
+  public void setLeft(double value) {
+    this.left.set(ControlMode.PercentOutput, value);
+  }
+
+  public void setRight(double value) {
+    this.right.set(ControlMode.PercentOutput, value);
+  }
+
+  @Override
+  public void periodic() {
+    for (ChickenTalon talon : masters) {
+      talon.config_kP(0, Tuning.drivetrainP);
+      talon.config_kI(0, Tuning.drivetrainI);
+      talon.config_kD(0, Tuning.drivetrainD);
+      talon.config_kF(0, Tuning.drivetrainF);
+      talon.config_IntegralZone(0, Tuning.drivetrainIZone);
+    }
+
+    for (ChickenTalon talon : talons) {
+      talon.configClosedloopRamp(Tuning.drivetrainRampRate);
+    }
+  }
+
+  public double getLeftPosition() {
+    return left.getSelectedSensorPosition();
```

Comment:
You can also directly grab the quadrature since we don't seem to do anything else anyways

Reply:
Why not be polymorphic?

Reply:
Accidents with grabbing the wrong sensor mostly if someone accidentally sets something; if we know that it needs to be from the quad, get it from the quad. Although sure that's fine if that's what we want.

Reply:
There's nothing software-wise _preventing_ it from being a different type of encoder, you'd just plug different values in. While defensive programming is a good thing generally, the feedback sensor can only be set within the same source file and if feedback sensors are getting set willy-nilly we have a much bigger problem.